src/librustdoc/formats/cache.rs

```diff
@@ -359,14 +361,19 @@ impl<'a, 'tcx> DocFolder for CacheBuilder<'a, 'tcx> {
                 // `public_items` map, so we can skip inserting into the
                 // paths map if there was already an entry present and we're
                 // not a public item.
-                if !self.cache.paths.contains_key(&item.def_id)
-                    || self.cache.access_levels.is_public(item.def_id)
+                if !self.cache.paths.contains_key(&item.def_id.expect_real())
+                    || self.cache.access_levels.is_public(item.def_id.expect_real())
                 {
-                    self.cache.paths.insert(item.def_id, (self.cache.stack.clone(), item.type_()));
+                    self.cache.paths.insert(
+                        item.def_id.expect_real(),
```

Comment:
I have a feeling this will ICE.

Reply:
Should `paths` also accept fake ids, or should we just check if it's real and only then insert it?

Reply:
Given that this didn't ICE documenting libstd, I'm ok with just leaving it as-is.